services/api/src/v1/__tests__/auth.js

```diff
@@ -92,7 +92,7 @@ describe('/1/users', () => {
         token: 'some bad token not really a good token'
       });
       expect(response.status).toBe(400);
-      expect(response.body.error.message).toBe('bad jwt token');
+      expect(response.text).toBe('bad jwt token');
```

Comment:
Why is returning a non-JSON error better? I don't understand.

I prefer it if the API would always return error envelopes in JSON. That way we can always parse and the UI can do things like take action based on type, or have more meta-data in there in case of form validation, etc.

Reply:
Right, sorry... I didn't meant to make a big opinionated change. Let me explain my thinking here:

I was noticing that in a few places we aren't being very consistent about error handling. There are a couple places like this but notably if I mess up the URL then every 404 that doesn't hit the router just returns plain text. Now we could go into the middleware and add more code to make sure it does, but there will always be a case where we can't return a JSON response which is a 500 when the server can't build one (even if we could the API could potentially be sitting behind CloudFlare or something else we can't guarantee the response of). Back in the JSONP days this would be a problem but these days clients can handle both, so we should make sure they do (given the above).

I totally agree that in many cases you need to return metadata on the error (like Joi validation failures) and we should definitely handle that too, but a lot of the time you just need to return a single error message and it seems simpler to default to that with the understanding that the client can handle either.

On a related note re: the custom Error types I added, I'm totally willing to get rid of those and always use `ctx.throw` but just to explain my thinking was (ironically) similar to your reason for disliking them which is "why should I have to know about `ctx.throw` when vanilla JS has error handling built in?". Thinking about it though, you're right that having to know about the different error types is overhead too. We could reduce this to being a single error type that just has a message and status code, but then it would essentially be the same thing as `ctx.throw` with the same overhead, so I'm willing to backpedal this one.